File: 0070-climbing-stairs/0070-climbing-stairs.kt
class Solution {
fun climbStairs(n: Int): Int {
    if (n == 1) return 1
    if (n == 2) return 2
    var first = 1
    var second = 2
    var third = 0
    for (i in 3..n) {
        third = first + second
        first = second
        second = third
    }
    return third
}
}

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let dp=[]
    dp[1]=1
    dp[2]=2
    
    for(let i=3;i<=46;i++){
        dp[i]=dp[i-1]+dp[i-2]
    }
    return dp[n]
    
};